### ISSUE TITLE
[Enhancement]: Allow for extending a volumes size

### ISSUE THREAD

**Issue by kieranbrown** [andrewbaxter/terraform-provider-fly#4]
It would be a nice enhancement if the `fly_volume` resource `size` parameter could be increased to allow for extending a volume (https://fly.io/docs/apps/volume-manage/#extend-a-volume).

I have a workaround which is to `ignore_changes` on the size and perform the extension through their API, but it would be a nice addition to the Terraform provider to support this.

Currently, the provider errors with the following:

```hcl
Terraform will perform the following actions:

  # fly_volume.this will be updated in-place
  ~ resource "fly_volume" "this" {
      ~ encrypted = true -> (known after apply)
      ~ id        = "vol_vd3y5n5ljd9dwjxr" -> (known after apply)
        name      = "data"
      ~ size      = 3 -> 4
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

fly_volume.this: Modifying... [id=vol_vd3y5n5ljd9dwjxr]
╷
│ Error: The fly api does not allow updating volumes once created
│ 
│   with fly_volume.this,
│   on main.tf line 20, in resource "fly_volume" "this":
│   20: resource "fly_volume" "this" {
│ 
│ Try deleting and then recreating a volume with new options
```

**Comment by andrewbaxter** [comment]
This seems pretty straightforward.  Right now it just isn't implemented, but there's an normal rest api for doing it: https://fly.io/docs/machines/api/volumes-resource/

**Comment by andrewbaxter** [comment]
I think I fixed this here: https://github.com/andrewbaxter/terraform-provider-fly/commit/7c92583732c7107a94b70ed6f282896cf59a1f6d (and hopefully didn't break anything else)

Released as 0.1.13 - let me know if it doesn't work or you have any issues with it!

**Comment by kieranbrown** [comment]
> I think I fixed this here: [7c92583](https://github.com/andrewbaxter/terraform-provider-fly/commit/7c92583732c7107a94b70ed6f282896cf59a1f6d) (and hopefully didn't break anything else)
> 
> Released as 0.1.13 - let me know if it doesn't work or you have any issues with it!

Yep, this is working flawlessly. I've also tested releasing IP addresses, which is working as expected.

Again, thanks for your hard work and quick turnaround on this 🎉

**Comment by andrewbaxter** [comment]
Nice! So I managed to accomplish at least one thing today :D
